x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Lookup.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.plan.logical;
+
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xpack.esql.expression.TableColumnAttribute;
+import org.elasticsearch.xpack.ql.capabilities.Resolvables;
+import org.elasticsearch.xpack.ql.expression.Attribute;
+import org.elasticsearch.xpack.ql.plan.logical.LogicalPlan;
+import org.elasticsearch.xpack.ql.plan.logical.UnaryPlan;
+import org.elasticsearch.xpack.ql.tree.NodeInfo;
+import org.elasticsearch.xpack.ql.tree.Source;
+
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.esql.expression.NamedExpressions.mergeOutputAttributes;
+
+/**
+ * Looks up values from the {@code tables} request parameter.
+ * <p>
+ *     This class exists in two very different modes - an "unresolved" mode
+ *     which contains only the {@link #tableName} and {@link #matchValues},
+ *     and a "resolved" mode which contains everything. The process of
+ *     resolving all of the references adds the extra columns by looking
+ *     up the table from the analysis context.
+ * </p>
+ */
+public class Lookup extends UnaryPlan {
+    private final Attribute tableName;
+    private final List<Attribute> matchFields;
+    private final List<TableColumnAttribute> matchValues;
+    private final List<TableColumnAttribute> mergeValues;
+    private List<Attribute> output;
+
+    public Lookup(
+        Source source,
+        LogicalPlan child,
+        Attribute tableName,
+        List<Attribute> matchFields,
+        @Nullable List<TableColumnAttribute> matchValues,
+        @Nullable List<TableColumnAttribute> mergeValues
+    ) {
+        super(source, child);
+        this.tableName = tableName;
+        this.matchFields = matchFields;
+        this.matchValues = matchValues;
+        this.mergeValues = mergeValues;
+    }
```

Comment:
Like Enrich this class tries to do two things:
- define the stiching/join
- act as a source (without being one) for the broadcast side.
This works but same limitations - let's discuss this offline.